pkg/dockercreds/k8svolume/provider.go

```diff
@@ -0,0 +1,77 @@
+package k8svolume
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"k8s.io/kubernetes/pkg/credentialprovider"
+)
+
+const (
+	SecretFilePathEnv = "CREDENTIAL_PROVIDER_SECRET_PATH"
+)
+
+func init() {
+	credentialprovider.RegisterCredentialProvider("secret-volume", &VolumeSecretProvider{})
+}
+
+type VolumeSecretProvider struct {
+}
+
+func (k VolumeSecretProvider) Enabled() bool {
+	_, ok := os.LookupEnv(SecretFilePathEnv)
+	return ok
+}
+
+func (k VolumeSecretProvider) Provide() credentialprovider.DockerConfig {
+	secretFolder := os.Getenv(SecretFilePathEnv)
+
+	dockerConfig, err := parseDockerConfigJson(filepath.Join(secretFolder, ".dockerconfigjson"))
```

Comment:
Is there a reason not to use `dockercreds.ParseDockerPullSecrets`?

Reply:
I decided not to use `ParseDockerPullSecrets` because we talked about supporting docker config in json only and the return types of the functions do not match, so we would have to create an adaptation layer.
We can create a different PR to standardize our types and start using the structs provided by k8s instead of handrolling `DockerCreds` or `dockerConfigJson`

Reply:
Ok. I don't know if would necessarily make sense to rely on k8s.DockerConfigEntry as it just has BasicAuth username/password keys.